gtsam/linear/JacobianFactor.cpp

```diff
@@ -220,60 +220,13 @@ FastVector<JacobianFactor::shared_ptr> _convertOrCastToJacobians(
 }
 
 /* ************************************************************************* */
-JacobianFactor::JacobianFactor(const GaussianFactorGraph& graph,
-    boost::optional<const Ordering&> ordering,
-    boost::optional<const VariableSlots&> p_variableSlots) {
-  gttic(JacobianFactor_combine_constructor);
-
-  // Compute VariableSlots if one was not provided
-  // Binds reference, does not copy VariableSlots
-  const VariableSlots & variableSlots =
-      p_variableSlots ? p_variableSlots.get() : VariableSlots(graph);
+void JacobianFactor::JacobianFactorHelper(const GaussianFactorGraph& graph,
+    const FastVector<VariableSlots::const_iterator>& orderedSlots) {
 
   // Cast or convert to Jacobians
   FastVector<JacobianFactor::shared_ptr> jacobians = _convertOrCastToJacobians(
       graph);
 
-  gttic(Order_slots);
```

Comment:
4 new constructors (generated from combinations of 2 optional parameters) will use this same code so i moved it into a separate function, not sure why diff util didn't recognize it better